### PR TITLE
Ac 128068203 index referrals

### PR DESCRIPTION
**Pull request by nate-cc**

### CODE REVIEW

File: app/helpers/referral_helper.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+# Referral utility module used to minimise presentation logic in views
+module ReferralHelper
+  def name_and_reference(referral)
+    "#{referral.name} - #{referral.reference}"
+  end
+end
```
Comment (pemacc) [inline app/helpers/referral_helper.rb]: Do we have a preference for Helper vs Presenter/Decorator?

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -1,4 +1,7 @@
 # frozen_string_literal: true
 
 module ApplicationHelper # :nodoc:
+  def format_date(value)
+    Date.parse(value).strftime('%m/%d/%Y') if value
+  end
 end
```
Comment (pemacc) [inline app/helpers/application_helper.rb]: Is this method going to be used in any other parts of the app other than referral?

File: app/views/referrals/index.html.haml [hunk 3]
```diff
@@ -1,24 +1 @@
...
-          %tr
-            %td
-              = link_to name_and_reference(referral), referral_path(referral)
-            %td
-              = ResponseTime::ALL[referral.response_time]
-            %td
-              = ScreeningDecision::ALL[referral.screening_decision]
-            %td
-              = format_date(referral.created_at)
+= render 'table', :referrals => @referrals
```
Comment (pemacc) [inline app/views/referrals/index.html.haml]: referals: @referrals

**Comment by pemacc** [comment]
Can you check why the test is failing? Left some comments?

File: app/helpers/application_helper.rb (hunk 2 above)
Comment (nate-cc) [inline app/helpers/application_helper.rb]: I assume so

File: app/helpers/referral_helper.rb (hunk 1 above)
Comment (nate-cc) [inline app/helpers/referral_helper.rb]: I don't think there is any reason to not use helpers

**Comment by nate-cc** [comment]
@pemacc I have moved referral_helper into referral_presenter, can you please review. Another option is to use Draper and assign the presenter in the controller but then we will have to delegate down. Any thoughts?

File: app/presenters/referral_presenter.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+# Referral presenter used to minimise presentation logic in views
+class ReferralPresenter
+  def initialize(referral)
+    @referral = referral
+  end
+
+  def name_and_reference
+    "#{@referral.name} - #{@referral.reference}"
+  end
+end
```
Comment (pemacc) [inline app/presenters/referral_presenter.rb]: Why not make it a class method which takes @referral object as argument. That way it an be used as 
ReferralPresenter.name_and_reference(referral),

**Pull request closed without merging**